### ISSUE TITLE
Restarting MPV Shim never works because of cred.json

### ISSUE THREAD

**Issue by xNinjaKittyx** [jellyfin/jellyfin-mpv-shim#331]
**Describe the bug**

Once MPV Shim closes, I'm never able to successfully come back up unless I manually go and delete conf.json and recreate it again.

This might be a separate issue, but I also see run.exe persisting after closing the application.

**To Reproduce**
1. Open MPV Shim
2. Input credentials
3. It works
4. Close MPV Shim
5. Restart MPV Shim
6. No window opens, only the tray opens.
7. Tray right click menu shows the options but none of them do anything.
8. Deleting conf.json and then letting mpv shim recreate it again will work.

**Expected behavior**
Restart should just work.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Version 2.6.0

**Error Messages**
Please provide logs, as they are often needed for me to understand and quickly troubleshoot the issue. You can read instructions for how to do so here:
https://github.com/iwalton3/jellyfin-mpv-shim/wiki/Sending-Logs

I don't really see any errors, but I do see this which is different than normal I think.

```
2023-03-31 22:08:10,515 [   DEBUG] urllib3.connectionpool: Starting new HTTPS connection (1): <redacted>:443
2023-03-31 22:08:10,531 [    INFO] websocket: Websocket connected
2023-03-31 22:08:10,531 [    INFO] JELLYFIN.jellyfin_apiclient_python.ws_client: --->[ websocket ]
2023-03-31 22:08:10,531 [   DEBUG] event_handler: Unhandled Event WebSocketConnect: None
```

Conf.json looks like this
```
[{"address": "https://<redacted>", "Name": "<redacted>", "Id": "<redacted>", "DateLastAccessed": "2023-03-09T22:16:08Z", "UserId": "<redacted>", "AccessToken": "<redacted>", "Users": [{"Id": "<redacted>", "IsSignedInOffline": true}], "uuid": "bf80797e-85ec-4a76-bbc4-2516d712db83", "username": "admin", "connected": true}]
```

**Comment by qvattr0** [comment]
I'm having the same issue. Were you able to find any alternative solutions to this bug?

**Comment by iwalton3** [comment]
Are you deleting conf.json or cred.json?

**Comment by qvattr0** [comment]
> Are you deleting conf.json or cred.json?

I deleted `conf.json` to fix the issue.

**Comment by iwalton3** [comment]
Each time you delete the config file it generates a new unique client id. I am honestly surprised it works at all after doing that.

**Comment by qvattr0** [comment]
> Each time you delete the config file it generates a new unique client id. I am honestly surprised it works at all after doing that.

I suppose that's strange. I've spent a really long time trying to debug MPV yesterday and I remember that uninstalling the software did not fix the problem. Only after deleted `conf.json` and re-adding my server was I able to have it work properly again. I've tried restarting my PC after fixing the issue (also quitting the application and relaunching it) to ensure that the issue does not arise again, and I've confirmed that it doesn't.

I'll include some of my observations from my experience yesterday to offer some potentially useful insight toward finding the root of this bug.
- As mentioned in the original post, the tray menu was the only operational entity left after MPV stopped working as intended. Something interesting I've noticed is that if you do actually try to "quit" MPV via the tray, the icon disappears from the tray but the process for MPV remains active (`run.exe` in Task Manager). Continuously starting and restarting MPV Shim resulted in having multiple instances of `run.exe` running in the background.
- What might've potentially caused the issue was me experimenting with `mpv.conf`. I was initially trying to install the MPV-OSC-Modern skin for the player, but have faced issues in the form of the original OSC never disappearing despite the `osc` flag being assigned a `no` in the config file. After realizing that the player was not removing the OSC, I started tweaking the various parameters/flags within the `mpv.conf` file. The only code there at the time was `vo=gpu-next` and some other code associated with enabling the aforementioned skin. I also started having the issue of HDR not working in the player (it worked in the first time I installed it, after adding `vo=gpu-next`) and then prompted further tweaking of the configuration file. Right now, I have the configuration file set properly (no tomfoolery involved in there) with code I found on the net for HDR and the thing's been running like a charm. I haven't embarked on the journey of installing the skin again but I'll do so once I have more free time. Now, I don't know how the code of the shim operates, but my best guess would be that tweaking the conf. in a certain way when I was experimenting with it killed the player. Furthermore, I believe that the configuration file is not wiped after an uninstall of the MPV player so reinstalling would have no impact on returning it to proper operation.

**Comment by iwalton3** [comment]
For OSC you need to disable the custom MPV Shim OSC by setting `thumbnail_osc_builtin` to false in `conf.json`.

**Comment by qvattr0** [comment]
> For OSC you need to disable the custom MPV Shim OSC by setting `thumbnail_osc_builtin` to false in `conf.json`.

Wait, the MPV Shim has a custom OSC? Do you mean the one in the picture I attached below? Given how "barebones" it looks, I thought it might be the default OSC.

![image](https://github.com/jellyfin/jellyfin-mpv-shim/assets/141742777/6acc88d3-b3f7-4c64-966b-c71fe38e72ce)

**Comment by iwalton3** [comment]
Yup it is custom. It is a very minimal fork which adds support for thumbfast thumbnails. Jellyfin MPV Shim comes with a thumbfast compatibility layer which can load chapter images or JellyScrub seek thumbnail previews from your Jellyfin server.

**Comment by qvattr0** [comment]
What does `osc=no` do then? Remove the menu (`c` key) from appearing?

**Comment by iwalton3** [comment]
`osc=no` disables the built in OSC provided by MPV, which is also the default. You can disable the OSC completely via `conf.json`. The MPV Shim OSC is loaded as a lua script since there is no way to override the MPV OSC otherwise.

**Comment by qvattr0** [comment]
I see, that makes sense. Thank you! I've got an unrelated issue that I've been having issues with (HDR in particular). Should I create a new issue or is this place fine for discussion?

**Comment by iwalton3** [comment]
Please open new issues for better tracking

**Comment by qvattr0** [comment]
Alright, I'll modify my previous comment and delete this one after a minute.

**Comment by PanoptesDreams** [comment]
Just experienced this. Fix is deleting conf.json in '%appdata%\jellyfin-mpv-shim'.

You will also need to remove the configured server and reconnect.

This is a long standing, persistent issue that has plagued me a lot in the past.

**Comment by G2COMMANDER** [comment]
As Izzie mentioned, the actual fix is to go into conf.json and change "enable_osc" to false. No need to delete or modify cred.json.
